primitives/src/opcodes.rs

```diff
@@ -71,6 +72,40 @@ macro_rules! all_opcodes {
                 }
             }
         }
+
+        impl str::FromStr for Opcode {
+            type Err = ();
+            fn from_str(s: &str) -> Result<Self, Self::Err> {
```

Comment:
In fd78f83e7e4f2fb6dcbd5ce8ca555b436973f72b:

I really don't like this fromstr impl ... it parses many ambiguous things. I'm not sure what we want to do here but I'm inclined to say "no parsing of opcodes from strings in rust-bitcoin" and make downstream users implement their own.

Reply:
@storopoli FYI we investigated this already recently in #3617.

Reply:
I think this is valid being defined here since it is impossible for a crate outside of `rust-bitcoin` to implement it (Rust's orphan rule).
And please note that `FromStr` returns a `Result` which the user must deal with it.
It is not that big of a footgun, as portrayed.

Reply:
I did add some tests that @tcharding had in #3617 with a typo fix that the second roundtrip (`Debug`) is `s2` and not `s1` again.